app/code/Magento/InventorySales/Test/Integration/IsProductSalableForRequestedQty/IsSalableWithReservationsConditionTest.php

```diff
@@ -0,0 +1,156 @@
+     * @var SourceItemRepositoryInterface
+     */
+    private $sourceItemRepository;
+
+    /**
+     * @var SearchCriteriaBuilder
+     */
+    private $searchCriteriaBuilder;
+
+    /**
+     * @var SourceItemsSaveInterface
+     */
+    private $sourceItemsSave;
+
+    /**
+     * @inheritdoc
+     */
+    protected function setUp()
+    {
+        parent::setUp();
+
+        $this->reservationBuilder = Bootstrap::getObjectManager()->get(ReservationBuilderInterface::class);
+        $this->appendReservations = Bootstrap::getObjectManager()->get(AppendReservationsInterface::class);
+        $this->cleanupReservations = Bootstrap::getObjectManager()->get(CleanupReservationsInterface::class);
+        $this->isProductSalableForRequestedQty
+            = Bootstrap::getObjectManager()->get(IsProductSalableForRequestedQtyInterface::class);
+        $this->productRepository = Bootstrap::getObjectManager()->get(ProductRepositoryInterface::class);
+        $this->stockItemRepository = Bootstrap::getObjectManager()->get(StockItemRepositoryInterface::class);
+        $this->stockItemCriteriaFactory = Bootstrap::getObjectManager()->get(
+            StockItemCriteriaInterfaceFactory::class
+        );
+        $this->sourceItemRepository = Bootstrap::getObjectManager()->get(SourceItemRepositoryInterface::class);
+        $this->searchCriteriaBuilder = Bootstrap::getObjectManager()->get(SearchCriteriaBuilder::class);
+        $this->sourceItemsSave = Bootstrap::getObjectManager()->get(SourceItemsSaveInterface::class);
+    }
+
+    /**
+     * @magentoDataFixture ../../../../app/code/Magento/InventoryApi/Test/_files/products.php
+     * @magentoDataFixture ../../../../app/code/Magento/InventoryApi/Test/_files/sources.php
+     * @magentoDataFixture ../../../../app/code/Magento/InventoryApi/Test/_files/stocks.php
+     * @magentoDataFixture ../../../../app/code/Magento/InventoryApi/Test/_files/stock_source_links.php
+     * @magentoDataFixture ../../../../app/code/Magento/InventoryApi/Test/_files/source_items.php
+     * @magentoDataFixture ../../../../app/code/Magento/InventoryIndexer/Test/_files/reindex_inventory.php
+     *
+     * @param string $sku
+     * @param int $stockId
+     * @param bool $isSalable
+     *
+     * @dataProvider productIsSalableDataProvider
+     */
+    public function testProductIsSalable(string $sku, int $stockId, float $qty, bool $isSalable)
+    {
+        self::assertEquals($isSalable, $this->isProductSalableForRequestedQty->execute($sku, $stockId, $qty));
+    }
+
+    /**
+     * @return array
+     */
+    public function productIsSalableDataProvider(): array
+    {
+        return [
+            ['SKU-1', 10, 1, true],
+            ['SKU-1', 20, 1, false],
+            ['SKU-1', 30, 1, true],
+            ['SKU-2', 10, 1, false],
+            ['SKU-2', 20, 1, true],
+            ['SKU-2', 30, 1, true],
+            ['SKU-3', 10, 1, false],
+            ['SKU-3', 20, 1, false],
+            ['SKU-3', 30, 1, false],
```

Comment:
why do we always make a check for Qty  = 1 ?

Reply:
the qty is never used because the tested methods are only proxy methods;
for that reason, let's discuss removing those tests at all

Reply:
as discussed in our call, we decided to leave the tests